### PR TITLE
Closes #2366 : Feature add sorting tabs data

### LINKED ISSUE #2366: [FEATURE] Milestones order change
### Description

we can change the order of milestones from older first to newer first so that we don't have to scroll through the website to see recent milestones 

or we can add a button at the top to sort them in ascending or descending order

although i am not clear with the idea love to see other people take in this

### Screenshots

Screenshot from 2022-12-10 19-59-08

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by prajapati55**
## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [x] My change requires changes to the documentation.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Note to reviewers

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/2474"></a>

### CODE REVIEW

**Review by Pradumnasaraf — CHANGES_REQUESTED** [review]
Hey, 
There is some merge conflict with your PR. Please resolve this so that we can merge it.

If you need any help feel free to reach out.

**Comment by eddiejaoude** [comment]
I think this PR needs updating with `main` branch

**Comment by prajapati55** [comment]
Screenshots
Events_Desc_By_Date
Milestones_Desc_By_Date
MyLinks_alphabatical_Desc_order
Testimonials_Desc_By_date

**Comment by prajapati55** [comment]
> comment
Resolved Conflicts.

File: styles/globals.css [hunk 1]
```diff
@@ -16,6 +16,13 @@ a {
 * {
   box-sizing: border-box;
 }
+.sortIcon {
```
Comment (eddiejaoude) [inline styles/globals.css:19]: Please keep styling relevant to the component

**Comment by eddiejaoude** [comment]
Looks great 👍  I will have a play today

**Comment by eddiejaoude** [comment]
You have conflicts with Changelog because you made changes to this file, this is an autogenerated file, please remove any changes

File: styles/globals.css (hunk 1 above)
Comment (prajapati55) [reply styles/globals.css:19]: ok.

**Comment by eddiejaoude** [comment]
Looks great 👍 , on smaller screen size it overlaps with the tab text

Screenshot 2022-12-29 at 15 54 16

File: components/user/UserTabs.js [hunk 2]
```diff
@@ -32,20 +33,36 @@ export default function UserTabs({ tabs, setTabs }) {
...
                     : "border-transparent text-gray-500 hover:text-gray-700 hover:border-gray-300",
-                  "w-1/4 py-4 px-1 text-center border-b-2 font-medium text-sm"
+                  "block py-4 px-1 text-center border-b-2 font-medium text-sm"
                 )}
                 aria-current={tab.current ? "page" : undefined}
               >
                 {tab.name} ({tab.total})
               </a>
+              {tab.current ? <BiSortAlt2
+              className="absolute sortIcon"
```
Comment (eddiejaoude) [inline components/user/UserTabs.js:51]: I think this class can be removed now? `sortIcon`

File: components/user/UserTabs.js [hunk 3]
```diff
@@ -32,20 +33,36 @@ export default function UserTabs({ tabs, setTabs }) {
...
                     ? "border-indigo-500 text-indigo-600"
                     : "border-transparent text-gray-500 hover:text-gray-700 hover:border-gray-300",
-                  "w-1/4 py-4 px-1 text-center border-b-2 font-medium text-sm"
+                  "block py-4 px-1 text-center border-b-2 font-medium text-sm"
                 )}
                 aria-current={tab.current ? "page" : undefined}
               >
                 {tab.name} ({tab.total})
               </a>
+              {tab.current ? <BiSortAlt2
```
Comment (eddiejaoude) [inline components/user/UserTabs.js:50]: I think `tab.current && <BiSortAlt2` would be easier to read and then the ` null` is now required

**Comment by eddiejaoude** [comment]
I think I have fixed the smaller screen issue by using flex instead of absolute

File: pages/[username].js [hunk 4]
```diff
@@ -100,28 +149,28 @@ export default function User({ users, data, BASE_URL }) {
       </Head>
 
       <Page>
-        <UserProfile data={data} BASE_URL={BASE_URL} />
+        <UserProfile data={userData} BASE_URL={BASE_URL} />
 
-        <UserTabs tabs={tabs} setTabs={setTabs} />
+        <UserTabs tabs={tabs} setTabs={setTabs} sortUserTabItems={sortUserTabItems} />
```
Comment (eddiejaoude) [inline pages/[username].js:154]: Would it be better to move `sortUserTabItems` to the `components/user/UserTabs.js` file?

**Review by eddiejaoude — APPROVED** [review]
Thank you 👍

**Pull request merged**